zerver/views/realm.py

```diff
@@ -85,4 +85,7 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
         do_set_realm_message_editing(realm, allow_message_editing, message_content_edit_limit_seconds)
         data['allow_message_editing'] = allow_message_editing
         data['message_content_edit_limit_seconds'] = message_content_edit_limit_seconds
+    if message_retention_days is 0:
```

Comment:
we generally use `==` for numbers, not `is`.  `is` is basically only used for `None`.